src/som/interpreter/LexicalScope.java

```diff
@@ -18,24 +18,54 @@
 
 public abstract class LexicalScope {
 
+  private final LexicalScope outer;
+
+  public LexicalScope(final LexicalScope outer) {
+    this.outer = outer;
+  }
+
+  public LexicalScope getContextualScope() {
+    return outer;
```

Comment:
might need to be `context`?
At least that's what the method name suggests?
Also `context` vs. `contextual`. Please stick with one.